### PR TITLE
feat(enrichment): add undocumented-init-params rule

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Adds an opt-in enrichment rule that flags classes whose `__init__` takes parameters but neither the class docstring nor `__init__` docstring has an `Args:` or `Parameters:` section. Previously docvet had no way to detect completely undocumented constructor interfaces.

- Add `_check_undocumented_init_params` check function with PEP 570 positional-only param support
- Add `require_init_params: bool = False` config field (opt-in) and `_VALID_ENRICHMENT_KEYS` entry
- Add 20 unit tests covering all 9 ACs, edge cases, cross-rule non-interference, and structural types
- Create rule reference page, rules.yml entry, enrichment.md/configuration.md/mkdocs.yml docs updates

fix(enrichment): correct positional-only param handling in undocumented-init-params

Test: `uv run pytest tests/unit/checks/test_undocumented_init_params.py -v`

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass
- [x] Lint passes
- [x] Types pass

### Review Focus
- PEP 570 positional-only param extraction uses `(posonlyargs + args)[1:]` to correctly handle `self` placement with `/` syntax

### Related
- Story 35.6
- Complements `missing-param-in-docstring` — mutually exclusive by design

### CODE REVIEW

File: tests/unit/checks/test_undocumented_init_params.py [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+    source = '''\
+    class Server:
+        """A server."""
+
+        def __init__(self, host):
+            self.host = host
+    '''
+    tree = ast.parse(textwrap.dedent(source))
+    config = EnrichmentConfig()  # Default: require_init_params=False
+    findings = check_enrichment(source, tree, config, "server.py")
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply tests/unit/checks/test_undocumented_init_params.py]: Addressed in latest push — all three `check_enrichment(source, ...)` calls in this file now pass `textwrap.dedent(source)` to match the dedented tree.
